### PR TITLE
fix: fetch profile from userinfo endpoint

### PR DESCRIPTION
**Pull request by markusahlstrand**
For vipps we want to fetch the profile from the userinfo endpoint as the id-token doesn't contain email and other good stuff.

### CODE REVIEW

File: src/authentication-flows/social.ts [hunk 1]
```diff
@@ -69,6 +70,54 @@ export interface socialAuthCallbackParams {
   code: string;
 }
 
+async function getProfile(token: TokenResponse) {
+  if (token.id_token) {
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:74]: If the server returns an id-token

File: src/authentication-flows/social.ts [hunk 2]
```diff
@@ -69,6 +70,54 @@ export interface socialAuthCallbackParams {
...
+      jti,
+      nonce,
+      email: emailRaw,
+      email_verified,
+      auth_time,
+      nonce_supported,
+      ...profileData
+    } = parseJwt(token.id_token);
+
+    if (!emailRaw) {
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:93]: And.. if the id-token returns an email.

**Comment by danielgent** [comment]
Is this superceeded by https://github.com/sesamyab/auth/pull/409 @markusahlstrand ? Can we close this? :thinking: 

You asked me to look at this yesterday but I'm guessing since then you redid it

**Comment by markusahlstrand** [comment]
This is fixed in another branch

**Pull request closed without merging**